kalliope/tts/googletts/googletts.py

```diff
@@ -43,35 +46,83 @@ def _generate_audio_file(self):
         .. raises:: FailToLoadSoundFile
         """
 
+        if (len(self.words) < 1):
+            logger.error("Googletts : Invalid input %s", self.words)
+            self.words = "I could not find an answer."
+
         # Prepare payload
-        payload = self.get_payload()
+        payloads = self.get_payloads()
 
-        # getting the audio
-        r = requests.get(TTS_URL, params=payload, stream=True, timeout=TTS_TIMEOUT_SEC)
-        content_type = r.headers['Content-Type']
+        i = 0
```

Comment:
Extract from line 56 to line 74 in a separate method ?

Also I think in this case i = payloads.size(), so you probably do not need to define this new variable i.
eg : for x in range(1, paylaods.size()):